src/Cryptol/REPL/Monad.hs

```diff
@@ -415,13 +415,18 @@ getPrompt  = mkPrompt `fmap` getRW
 getCallStacks :: REPL Bool
 getCallStacks = eCallStacks <$> getRW
 
+clearTCSolverAction :: REPL (IO ())
+clearTCSolverAction =
+  REPL (\ref -> pure (modifyIORef ref (\rw -> rw { eTCSolver = Nothing })))
```

Comment:
Perhaps we should use `atomicModifyIORef` here, since this may occur asynchronously?